getnacelle/nacelle-js
# feat(gatsby-source-nacelle): handle nested content references

## WHY are these changes introduced?

Addresses [ENG-4700](https://nacelle.atlassian.net/browse/ENG-4700) 

- Replaces nested content nodes with links so Gatsby can link them to their top level nodes. This also allows arrays of content references to have different fields, which was broken before. 


## WHAT is this pull request doing?

- During source nodes, searches for arrays/objects inside the `remoteFields` key of content & replaces any object with a `nacelleEntryId` with a `___NODE` reference to the content node with the same Gatsby Id. For arrays, replaces the entire array with an array of Gatsby Node Id's. This allows for users to query their inferred, space-specific NacelleContent types using interface fragments. 

### Assumptions/Caveats

- Currently treats all objects with `nacelleEntryId` keys as if they are Content references.
- For custom ingestion/other sources, this won't handle arrays of disparate types that don't have nacelleEntryId's. As long as they don't have field name overlap with different types, Gatsby's native inference should handle those cases okay.  



## How to Test

1. Navigate to `packages/gatsby-source-nacelle` in your terminal
2. Run `npm link`
3. Navigate to either `starters/gatsby` or an existing/newly created Nacelle Gatsby Repo
   - Note that it's helpful to use a space with more complex content models that has internal content links.
4. Run `npm link @nacelle/gatsby-source-nacelle` to link to the version in this PR
5. Run `npm run develop` to start the dev server
6. Navigate to http://localhost:8000/___graphql and explore the queries for various content types
   - For arrays of content, should be able to query each of the content types included in the array using `...on NacelleContentRemote<TypeName>` fragment queries.
